chromium_src/third_party/blink/renderer/core/html/html_permission_element.cc

```diff
@@ -0,0 +1,14 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "third_party/blink/renderer/core/html/html_permission_element.h"
+
+#include "third_party/blink/public/mojom/permissions/permission.mojom-blink.h"
+#include "third_party/blink/renderer/core/dom/document.h"
+#include "third_party/blink/renderer/core/dom/element.h"
+
+#define PermissionStatus mojom::blink::PermissionStatus
```

Comment:
what is the conflict? Why do we have a conflict here? Can you add comments?

Reply:
cc: @cdesouza-chromium

Reply:
I get:
```
../../third_party/blink/renderer/core/html/html_permission_element.cc(99,55): error: reference to 'PermissionStatus' is ambiguous
   99 | int GetMessageIDSinglePermission(PermissionName name, PermissionStatus status) {
      |                                                       ^
../../third_party/blink/renderer/core/html/html_permission_element.cc(37,21): note: candidate found by name lookup is 'blink::PermissionStatus'
   37 | using mojom::blink::PermissionStatus;
      |                     ^
../..\third_party/blink/renderer/bindings/core/v8/active_script_wrappable_creation_key.h(84,16): note: candidate found by name lookup is 'blink::PermissionStatus'
   84 |   friend class PermissionStatus;
      |                ^
```

Reply:
This is a fairly complex issue to explain, but the main problem is that there's another type called `PermissionStatus` in another file in blink, and due to the way we do our overrides, that file ends up being pulled in the build of `html_permission_element.cc`, and causing an ambiguity. I should have left a comment in the code explaining that.

I can add a comment explaining that. This change also required the introduction of patch file, to stop `patches/third_party-blink-renderer-core-html-html_permission_element.cc.patch`, which removes the aliasing in the blink file.

```
+-using mojom::blink::PermissionStatus;
```